### DISCUSSION TITLE
Language detection not working (linguist-vendored=true ignored)

### DISCUSSION THREAD

**Discussion by ibc** [community/community#42654] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Language detection definitely doesn't update properly in my project [mediasoup](https://github.com/versatica/mediasoup). 

As you can see, it keeps saying that there is 10.1% of Perl code in [`worker/deps` folder](https://github.com/versatica/mediasoup/search?l=perl):
<img width="376" alt="CleanShot-2022-12-26-at-13 21 11@2x" src="https://user-images.githubusercontent.com/16191/209548465-4889832d-02c0-4be9-84f7-a11e3f6f79c0.png">
<img width="679" alt="CleanShot-2022-12-26-at-13 23 49@2x" src="https://user-images.githubusercontent.com/16191/209548655-6726f698-7747-43a2-94cc-7994a9e98388.png">

However `worker/deps` is ignored in [`.gitattributes`](https://github.com/versatica/mediasoup/blob/v3/.gitattributes) file in project's root folder:

```.gitignore

# Tell GitHub to ignore some folders when detecting the project language.
art/* linguist-vendored=true
doc/* linguist-vendored=true
rust/benches/* linguist-vendored=true
rust/examples/* linguist-vendored=true
rust/examples-frontend/* linguist-vendored=true
rust/tests/* linguist-vendored=true
rust/**/tests.rs linguist-vendored=true
worker/deps/* linguist-vendored=true
worker/subprojects/* linguist-vendored=true
worker/test/* linguist-vendored=true
worker/scripts/* linguist-vendored=true
worker/fuzzer/* linguist-vendored=true
worker/out/* linguist-vendored=true
```

This is like this for months already. It never auto-fixes.

**Comment by jfullstackdev** [comment]
I don't know if that is really accurate or there is a problem,

but here, maybe you can ask them

https://github.com/github/linguist

**Reply by ibc** [reply]
Not sure what to ask to [/linguist](https://github.com/github/linguist) guys. Obviously I'm following their docs and this is supposed to work (I confirm that it used to work in the past). Problem I see is that this is not taking any effect on GitHub side. I don't know how GH applies linguist into projects to show detecte languages.

**Comment by look** [comment, marked as answer]
Hi @ibc, thanks for your report. Sorry, but at this time, code search does not use the `.gitattributes` file for language detection, which is why it's not affecting your results.

**Reply by ibc** [reply]
Thanks. May I know how to configure which files/folder should it inspect or ignore?

**Reply by look** [reply]
Can you clarify what you mean by that? Thanks.

**Reply by ibc** [reply]
I mean how to tell code search to ignore some folders (such as the worker/deps folder in my project).

**Reply by look** [reply]
Got it. This is not currently possible. If you're in the new [Code Search beta](https://github.com/features/code-search) you can exclude paths from your search by using a regex like this: `NOT path:/worker\/deps/`
